types/cucumber/cucumber-tests.ts

```diff
@@ -152,6 +150,14 @@ function StepSampleWithoutDefineSupportCode() {
         assert.deepEqual(actual, expected);
     });
 
+    Given('Common world constructor', function() {
```

Comment:
in this case it looks like there is no inference source for `T` and it'll just default to `World`, right? If there's no usage of it afterward (like in the return type) then you can accomplish the same thing by using a `this: World` instead of `this: T` in `Given`. Then for the custom case, you just put `this: CustomWorld`.

I noticed that the no-unnecessary-generics rule is turned off in tslint.json for cucumber. If it were turned on it would have errored on this change.

Reply:
Done

Reply:
You might still want to have the `this: CustomWorld` test to make sure it continues to work.